src/main/scala/ch/openolitor/util/jsonpath/functions/JsonPathFunction.scala

```diff
@@ -180,15 +180,42 @@ object JsonPathFunctions {
    */
   object GroupBy extends Param1JsonPathFunction {
     def evaluate(property: String, jsValue: Vector[JsValue]): Option[Vector[JsValue]] = {
-      val groups = jsValue.groupBy(jsValue => JsonPath.query("$."+property, jsValue).right.toOption.flatMap {
+      val groups = jsValue.groupBy(jsValue => JsonPath.query("$." + property, jsValue).right.toOption.flatMap {
         // if property was not found, JsonPath evaluated to empty vector, map this case to None
         case Vector() => None
-        case x => Some(x)
+        case x        => Some(x)
       })
       // filter out not matched properties and map vector into a jsarray
       val result = groups.filterNot(_._1.isEmpty).map(entries => JsArray(entries._2)).toVector
       logger.debug(s"Result of $$groupBy function is:$result")
       Some(result)
     }
   }
-}
\ No newline at end of file
+
+  /**
+   * Pretty-Print the current json-tree as a way of debugging into the document
+   */
+  object Debug extends UnaryJsonPathFunction {
+    def evaluate(jsValues: Vector[JsValue]): Option[Vector[JsValue]] = {
+      val result = JsArray(jsValues).prettyPrint
+      Some(Vector(JsString(result)))
+    }
+  }
+
+  object Debug1Param extends Param1JsonPathFunction {
+    def evaluate(property: String, jsValues: Vector[JsValue]): Option[Vector[JsValue]] = {
+      val evaluated = jsValues map { jsValue: JsValue =>
```

Comment:
We don't need to re-resolve properties as they where resolved before calling the param1 function. You can directly prettify the jsValues

Reply:
Ah yes, this is true.
BTW: prettifying does not work, the string is printed on one line into the field (since fields are not multiline)

Reply:
Hmm... just had a look. The `BaseParam1StringFunction` calls the `extractNumber` function. And this is not what we need.

Reply:
No, but in line 209, the param get already resolved. So the provided jsValues should already be the result of evaluating the JsonPath

Reply:
Now I get it. It must be 
`Some(Vector(JsArray(evaluated)))`

Reply:
IMO directly `Some(Vector(JsArray(jsValues).prettyPrint))`, with the need to re-evauate

Reply:
What about the `property`? It won't be used...
As I pointed out, this is no `BaseParam1StringFunction` object where the evaluation has taken place beforehand.

We better discuss this in person :)